Bug: Type mismatch when using recommended config in defineConfig

### 🧩 Description

Hey Marco, thank you very much for your plugin!
There seems to be a slight issue when using your recommended config in a type-checked `eslint.config.js` file.

### 🪜 How to reproduce

1. `git clone https://github.com/ulrichstark/eslint-plugin-zod-x-config-type-repro`
2. `cd eslint-plugin-zod-x-config-type-repro`
3. `npm i`
4. `npx tsc`

### ✅ Expected behavior

No errors.

### 💻 System info

```shell
System:
    OS: macOS 26.0.1
    CPU: (10) arm64 Apple M4
    Memory: 3.27 GB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.9.0 - /Users/ulrichstark/.nvm/versions/node/v24.9.0/bin/node
    npm: 11.6.0 - /Users/ulrichstark/.nvm/versions/node/v24.9.0/bin/npm
    pnpm: 10.12.2 - /opt/homebrew/bin/pnpm
    bun: 1.2.16 - /Users/ulrichstark/.bun/bin/bun
    Deno: 2.4.5 - /Users/ulrichstark/.deno/bin/deno
  npmPackages:
    eslint: ^9.37.0 => 9.37.0 
    eslint-plugin-zod-x: ^1.5.0 => 1.5.0
```

### ✒️ Additional context

But the actual behavior is following TypeScript error when running `npx tsc`:

```
eslint.config.js:4:29 - error TS2345: Argument of type 'Config' is not assignable to parameter of type 'InfiniteArray<ConfigWithExtends>'.
  Type 'Config' is not assignable to type 'ConfigWithExtends'.
    Types of property 'languageOptions' are incompatible.
      Type 'import("/Users/ulrichstark/Desktop/eslint-plugin-zod-x-config-type-repro/node_modules/@typescript-eslint/utils/dist/ts-eslint/Config").FlatConfig.LanguageOptions | undefined' is not assignable to type 'import("/Users/ulrichstark/Desktop/eslint-plugin-zod-x-config-type-repro/node_modules/@eslint/core/dist/cjs/types").LanguageOptions | undefined'.
        Type 'import("/Users/ulrichstark/Desktop/eslint-plugin-zod-x-config-type-repro/node_modules/@typescript-eslint/utils/dist/ts-eslint/Config").FlatConfig.LanguageOptions' is not assignable to type 'import("/Users/ulrichstark/Desktop/eslint-plugin-zod-x-config-type-repro/node_modules/@eslint/core/dist/cjs/types").LanguageOptions'.
          Index signature for type 'string' is missing in type 'LanguageOptions'.

4 export default defineConfig(eslintPluginZodX.configs.recommended);
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in eslint.config.js:4
```

--- a/.changeset/weak-candles-dance.md
+++ b/.changeset/weak-candles-dance.md
@@ -1,0 +1,5 @@
+---
+'eslint-plugin-zod-x': patch
+---
+
+fix: plugin type is incompatible with `eslint#defineConfig`

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
-import type { TSESLint } from '@typescript-eslint/utils';
+import type { ESLint, Linter, Rule } from 'eslint';
 
 import { PLUGIN_NAME, PLUGIN_VERSION } from './meta.js';
 import { arrayStyle } from './rules/array-style.js';
@@ -18,7 +18,6 @@ const eslintPluginZodX = {
     name: PLUGIN_NAME,
     version: PLUGIN_VERSION,
   },
-  configs: {} as { recommended: TSESLint.FlatConfig.Config },
   rules: {
     /* eslint-disable @typescript-eslint/naming-convention */
     'array-style': arrayStyle,
@@ -33,15 +32,15 @@ const eslintPluginZodX = {
     'require-error-message': requireErrorMessage,
     'require-schema-suffix': requireSchemaSuffix,
     /* eslint-enable @typescript-eslint/naming-convention */
-  } as Record<string, TSESLint.LooseRuleDefinition>,
-} as const satisfies TSESLint.FlatConfig.Plugin;
+  } as unknown as Record<string, Rule.RuleModule>,
+} as const;
 
-eslintPluginZodX.configs.recommended = {
+const recommendedConfig: Linter.Config = {
   name: `${PLUGIN_NAME}/recommended`,
   files: ['**/*.{js,mjs,cjs,jsx,mjsx,ts,tsx,mtsx}'],
   /* eslint-disable @typescript-eslint/naming-convention */
   plugins: {
-    'zod-x': eslintPluginZodX,
+    'zod-x': eslintPluginZodX as ESLint.Plugin,
   },
   rules: {
     'zod-x/array-style': 'error',
@@ -57,4 +56,13 @@ eslintPluginZodX.configs.recommended = {
   /* eslint-enable @typescript-eslint/naming-convention */
 };
 
-export default eslintPluginZodX;
+export default {
+  ...eslintPluginZodX,
+  configs: {
+    recommended: recommendedConfig,
+  },
+} satisfies ESLint.Plugin;
+/**
+ * why `satisfies`?
+ * @see https://github.com/marcalexiei/eslint-plugin-zod-x/issues/49
+ */

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -15,6 +15,9 @@
     "module": "NodeNext",
     "target": "ES6",
 
+    // Emit
+    "removeComments": true,
+
     // Language and Environment
     "moduleResolution": "NodeNext",
 

